app.rb

```diff
@@ -0,0 +1,97 @@
+require_relative './student'
+require_relative './teacher'
+require_relative './book'
+require_relative './classroom'
+require_relative './rental'
+
+class App
+  attr_accessor :books, :students, :teachers, :rentals
+
+  def initialize
+    @books = []
+    @people = []
+    @rentals = []
+  end
+
+  def display_all_books
+    @books.each_with_index do |book, index|
+      puts "#{index}) Title: '#{book.title}', Author: '#{book.author}'"
+    end
+  end
+
+  def display_all_people
+    @people.each_with_index do |people, index|
+      puts "#{index} Name: #{people.name}, ID: #{people.id}, Age: #{people.age}"
+    end
+  end
+
+  def create_person
+    print 'Do you want to create a student(1) or a teacher(2)? [Input the number]: '
+    person_option = gets.chomp.to_i
+    print 'Age: '
+    age = gets.chomp.to_i
+    print 'Name: '
+    name = gets.chomp
+    if person_option == 1
+      print 'Classroom: '
+      classroom_ = gets.chomp
+      print 'Has parent permmission? [Y/N]: '
+      parent_permmission = gets.chomp.downcase == 'y'
+      create_student(name, age, parent_permmission, classroom_)
+    else
+      print 'Specialization: '
+      specialization = gets.chomp
+      create_teacher(name, age, specialization)
+    end
+    puts 'Person created successfully'
+  end
+
+  def create_student(name, age, parent_permmission, classroom)
+    classroom = Classroom.new(classroom)
+    student = Student.new(classroom, age, name, parent_permmission)
```

Comment:
- [ ] I was unable to create a student because you failed to pass in the value of `parent_permmission` well. Since you are using the colon notation, you need to declare the argument differently.

  ```
  student = Student.new(classroom, age, name, parent_permmission: parent_permmission)
  ```

  ![image](https://user-images.githubusercontent.com/36542986/200127921-b93dd0b6-23a2-4699-9141-d8d1ba5720f4.png)

  NOTE: You also need to do the same in the Student class (see comment below)